From: Alexander Schmidt <alexs@linux.vnet.ibm.com>
Subject: ib/ehca: Do not turn off irqs in tasklet context
References: bnc#575073,LTC#59985
Patch-mainline: Submitted - Wed, 27 Jan 2010 09:03:36 -0800 to linux-rdma@vger.kernel.org

The irq_spinlock is only taken in tasklet context, so it is safe not to
disable hardware interrupts.

Signed-off-by: Alexander Schmidt <al...@linux.vnet.ibm.com>
Acked-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/ehca/ehca_irq.c |    5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/infiniband/hw/ehca/ehca_irq.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/infiniband/hw/ehca/ehca_irq.c
+++ linux-2.6.32-SLE11-SP1/drivers/infiniband/hw/ehca/ehca_irq.c
@@ -548,11 +548,10 @@ void ehca_process_eq(struct ehca_shca *s
 	struct ehca_eq *eq = &shca->eq;
 	struct ehca_eqe_cache_entry *eqe_cache = eq->eqe_cache;
 	u64 eqe_value, ret;
-	unsigned long flags;
 	int eqe_cnt, i;
 	int eq_empty = 0;
 
-	spin_lock_irqsave(&eq->irq_spinlock, flags);
+	spin_lock(&eq->irq_spinlock);
 	if (is_irq) {
 		const int max_query_cnt = 100;
 		int query_cnt = 0;
@@ -643,7 +642,7 @@ void ehca_process_eq(struct ehca_shca *s
 	} while (1);
 
 unlock_irq_spinlock:
-	spin_unlock_irqrestore(&eq->irq_spinlock, flags);
+	spin_unlock(&eq->irq_spinlock);
 }
 
 void ehca_tasklet_eq(unsigned long data)
